Feature/#13 대회 crud api 구현

myeowon opened the pull request:
## 🔥 연관된 이슈

close: #13 

## 📜 작업 내용

- 대회 CRUD API를 개발했습니다.
  - 대회 생성
  - 대회 전체 조회
  - 대회 수정
  - 대회 삭제

## 💬 리뷰 요구사항

- 리뷰 요구사항을 작성해주세요.

## ✨ 기타

- 브랜치 이름에 대소문자가 혼용되어 동일 이슈에 대한 PR을 중복으로 올려버렸습니다.ㅜㅜ #16 PR은 삭제 요청해두었습니다!

myeowon committed:
feat : 대회 컨트롤러 구현

myeowon committed:
feat : 대회 수정

myeowon committed:
feat : 대회 삭제

myeowon committed:
feat : 전체 대회 조회

myeowon committed:
chore: PR 정리용 불필요 파일 삭제

myeowon committed:
feat : 대회 등록 시 대회 정보 반환

myeowon committed:
fix : 컨트롤러 어노테이션 수정

pykido reviewed (approved):
수고하셨습니다!

JJimini commented on src/main/java/com/opus/opus/modules/contest/application/ContestCommandService.java:
```diff
@@ -8,4 +16,34 @@
 @Transactional
 @RequiredArgsConstructor
 public class ContestCommandService {
+
+    private final ContestRepository contestRepository;
+    private final ContestConvenience contestConvenience;
```
nit : Repository와 Convenience 사이에 공백 넣어주면 구분하기 더 쉬울 것 같습니당

JJimini commented on src/main/java/com/opus/opus/modules/contest/application/ContestQueryService.java:
```diff
@@ -8,4 +14,27 @@
+                            category.getCategoryName(),
+                            contest.getIsCurrent(),
+                            contest.getUpdatedAt()
+                    );
```
정적 팩토리 메서드를 사용하는 여원님의 기준이 궁금합니다~! 이전 PR에서는 사용하셨던 것 같은데 이번에는 사용하지 않아서요!

myeowon replied:
정적 팩토리 메서드 사용하는 것을 통일하겠습니다!

JJimini commented on src/main/java/com/opus/opus/modules/contest/application/dto/response/ContestResponse.java:
```diff
@@ -0,0 +1,24 @@
+        Boolean isCurrent,
+        LocalDateTime updatedAt
+) {
+    public static ContestResponse from(Contest contest, String categoryName) {
+        return new ContestResponse(
+                contest.getId(),
+                contest.getContestName(),
+                contest.getCategoryId(),
+                categoryName,
+                contest.getIsCurrent(),
+                contest.getUpdatedAt()
+        );
+    }
```
실제 branch 내 코드에는 없는데 여기에는 있네요..?
????

JJimini reviewed (approved):
수고하셨습니다~! 이전 pr과 변경사항이 겹치는 부분이 좀 있는 것 같은데 체크 잘 해서 머지해주세용!

myeowon committed:
fix : 계행 추가, 정적 팩토리 메서드 사용으로 통일

myeowon committed:
feat : 대회 등록

# Conflicts:
#	src/main/java/com/opus/opus/modules/contest/application/convenience/ContestConvenience.java

myeowon committed:
fix : 대회 카테고리 요청
